File: impls/cpp2/step0_repl.cpp
#include <iostream>
#include <string>


const std::string prompt="user> ";

std::string READ(const std::string& input) {
  return input;
}

std::string EVAL(const std::string& input) {
  return input;
}

std::string PRINT(const std::string& input) {
  return input;
}

std::string rep(const std::string& input) {
  return PRINT(EVAL(READ(input)));
}

int main(int argc, char *argv[]) {

  std::cout << prompt << "\n";

  std::string line;
  while(std::getline(std::cin, line)) {
    std::cout << prompt << rep(line) << "\n";
  }

  std::cout << std::endl;
  return 0;
}

File: impls/myPython/step0_repl.py
#!/usr/bin/python3

def READ(line):
    return line

def EVAL(string):
    return string

def PRINT(string):
    return string

def rep(line):
    return PRINT(EVAL(READ(line)))

if __name__ == "__main__":
    while True:
        try: 
            line = input('user> ')
        except EOFError:
            break

        print(rep(line))

